src/utils.js

```diff
@@ -1,3 +1,46 @@
+const NON_STRING_PLACEHOLDER = "__REACT_EMOJI_RENDER_PLACEHOLDER__";
+
 export function escapeStringToBeUsedInRegExp(string) {
   return string.replace(/[.*+?^${}()|[\]\\]/g, "\\$&"); // $& means the whole matched string
 }
+
+export function stripNonStringElements(children) {
+  if (Array.isArray(children)) {
+    const stripped = children
+      .map(x => (typeof x === "string" ? x : NON_STRING_PLACEHOLDER))
+      .join("");
+    return [stripped, children.filter(x => typeof x !== "string")];
+  } else if (typeof children === "string") {
+    return [children, []];
+  }
+}
+
+export function returnStrippedElements(stripped, elements) {
+  if (!stripped || !Array.isArray(stripped)) return stripped;
```

Comment:
Same issue here I believe, `stripped` might be of other types than `Array` or `string` ?